incubator/hnc/internal/reconcilers/object.go

```diff
@@ -699,6 +705,26 @@ func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstruct
 	}
 }
 
+// getNoneSelector returns true indicates that user do not want this object to be propagated
+func (r *ObjectReconciler) getNoneSelector(log logr.Logger, inst *unstructured.Unstructured) bool {
+	annot := inst.GetAnnotations()
+	noneSelectorStr, ok := annot[api.AnnotationNoneSelector]
+	if !ok {
+		return false
+	}
+	// Empty string is treated as 'false'
+	if noneSelectorStr == "" {
+		return false
+	}
+	noneSelector, err := strconv.ParseBool(noneSelectorStr)
+	if err != nil {
+		// TODO: surface the error
+		log.Error(err, "Invalid noneSelector value, it should be either true or false.")
```

Comment:
Does `err` include the actual value of `noneSelectorStr`? If not, we should add it to the log.

Reply:
added